### PR TITLE
[fix-5430]: Centralize useDeviceMode hook

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-5430](https://gemeente-amsterdam.atlassian.net/browse/SIG-5430)

While fixing this bug I centralized the useDeviceMode hook so it's globally available and easier in use.

To fix the bug I added some focusTrapOptions when on mobile.

### CODE REVIEW

File: src/hooks/useDeviceMode.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2022 - 2023 Gemeente Amsterdam
+import { useEffect, useState } from 'react'
+
+import { sizes } from '@amsterdam/asc-ui/lib/theme/default/breakpoints'
+
+export enum DeviceMode {
```
Comment (OscarBakker) [inline src/hooks/useDeviceMode.ts:7]: Pin is using this enum as well. should this enum be extracted to a higher level?

File: src/hooks/useDeviceMode.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export interface ModeProp {
+  $mode: DeviceMode
+}
+
+function getDeviceMode(size: number) {
+  if (size <= sizes.tabletM) {
+    return DeviceMode.Mobile
+  } else {
+    return DeviceMode.Desktop
```
Comment (OscarBakker) [inline src/hooks/useDeviceMode.ts:20]: unnecessary return

**Review by OscarBakker — COMMENTED** [review]
Good job

File: src/hooks/useDeviceMode.ts (hunk 1 above)
Comment (vdegraaf) [reply src/hooks/useDeviceMode.ts:7]: good one. I put is in `types/device' on toplevel

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**